controller/users.go

```diff
@@ -580,6 +580,33 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 			}
 		}
 
+		updatedFeatureLevel := ctx.Payload.Data.Attributes.FeatureLevel
+		if log.IsDebug() {
+			currentFeatureLevel := "none"
+			newFeatureLevel := "none"
+			if user.FeatureLevel != nil {
+				currentFeatureLevel = *user.FeatureLevel
+			}
+			if updatedFeatureLevel != nil {
+				newFeatureLevel = *updatedFeatureLevel
+			}
+			log.Debug(ctx, map[string]interface{}{"current_feature_level": currentFeatureLevel, "new_feature_level": newFeatureLevel}, "updating feature level")
+		}
+		if updatedFeatureLevel != nil && (user.FeatureLevel == nil || *updatedFeatureLevel != *user.FeatureLevel) {
+			// handle the case where the value needs to be reset, when the new value is "" (empty string) or "released"
+			if *updatedFeatureLevel == "" || *updatedFeatureLevel == "released" {
+				log.Debug(ctx, map[string]interface{}{"user_id": user.ID}, "resetting feature level")
+				user.FeatureLevel = nil
+			} else {
+				// if the level is 'internal', we need to check against the email address to verify that the user is a Red Hat employee
+				if *updatedFeatureLevel == "internal" && !strings.HasSuffix(user.Email, "@redhat.com") {
+					log.Error(ctx, map[string]interface{}{"user_id": user.ID}, "user is not a Red Hat employee")
+					return errors.NewBadParameterError("feature_level", updatedFeatureLevel)
```

Comment:
Or rather 401 or 403?

Reply:
Right: `400` means that the request syntax is invalid and `401` means that the request did not contain the required credentials, which are not the cases here. 

`403` means that server understood the request but refuses to process it. That sounds like a better status, indeed.

Let me fix that